### PR TITLE
[Scaladoc] scala.util.Using code samples are a bit simplified

### PR DESCRIPTION
**Pull request by joshlemer**
* The concept of unfolding an iterator from immutable state would probably be considered intermediate-level functional programming, so using more simple Iterator operations is probably preferable in the scaladocs

* Required imports are also included in the example code

### CODE REVIEW

File: src/library/scala/util/Using.scala [hunk 1]
```diff
@@ -38,15 +41,19 @@ import scala.util.control.{ControlThrowable, NonFatal}
...
   *   val r2 = use(new BufferedReader(new FileReader("file2.txt")))
   *   val r3 = use(new BufferedReader(new FileReader("file3.txt")))
   *   val r4 = use(new BufferedReader(new FileReader("file4.txt")))
   *
   *   // use your resources here
-  *   def lines(reader: BufferedReader): Iterator[String] =
-  *     Iterator.unfold(())(_ => Option(reader.readLine()).map(_ -> ()))
+  *   def lines(reader: BufferedReader): Iterator[String] = {
+  *     Iterator.continually(reader.readLine()).takeWhile(_ != null)
+  *   }
```
Comment (som-snytt) [inline src/library/scala/util/Using.scala]: Recommend omitting braces, as it sets a style precedent and adds a line to doc.

**Review by som-snytt — APPROVED** [review]
I think this was the example I actually forwarded to a colleague. I've tried to use `Using` a few times and had to consult the doc as often as I do for `String`. I think the weird part is all the parens.

I'm on the fence about showing imports. I'd like a button to "copy example code to clipboard" which would include imports that weren't shown.

File: src/library/scala/util/Using.scala (hunk 1 above)
Comment (joshlemer) [reply src/library/scala/util/Using.scala]: oh yeah, you're right

Comment (joshlemer) [reply src/library/scala/util/Using.scala]: Fixed

**Comment by joshlemer** [comment]
@som-snytt this can be merged whenever you have a chance

**Pull request merged**